pkg/proxy/iptables/proxier.go

```diff
@@ -1002,7 +998,82 @@ func (proxier *Proxier) syncProxyRules() {
 		// Service does not have conflicting configuration such as
 		// externalTrafficPolicy=Local.
 		allEndpoints = proxy.FilterEndpoints(allEndpoints, svcInfo, proxier.nodeLabels)
-		hasEndpoints := len(allEndpoints) > 0
+
+		// Scan the endpoints list to see what we have. "hasEndpoints" will be true
+		// if there are any usable endpoints for this service anywhere in the cluster.
+		var hasEndpoints, hasLocalReadyEndpoints, hasLocalServingTerminatingEndpoints bool
```

Comment:
since these variables are just used to calculate `useTerminatingEndpoints` can we move this into another function? just for understandability

Reply:
I think we should do some careful refactoring of all this code (I started some in https://github.com/kubernetes/kubernetes/pull/106158) but we can do those as small followups.  Balancing readability of this code with the desire to DRY is hard.